mlflow/xgboost.py

```diff
@@ -286,6 +287,30 @@ def autolog(importance_types=["weight"]):  # pylint: disable=W0102
     import xgboost
     import numpy as np
 
+    class _InputExampleInfo:
+        def __init__(self, input_example=None, error_msg=None):
+            self.input_example = input_example
+            self.error_msg = error_msg
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Can we add documentation to this function explaining why we're patching it (i.e. in order to fetch an input example).